python_modules/dagster/dagster/_config/structured_config/__init__.py

```diff
@@ -900,8 +900,14 @@ def _convert_pydantic_field(pydantic_field: ModelField) -> Field:
             config_type=inferred_field.config_type,
             key_type=key_type,
         )
+        if pydantic_field.allow_none:
+            wrapped_config_type = Noneable(wrapped_config_type)
 
-        return Field(config=wrapped_config_type, description=inferred_field.description)
+        return Field(
+            config=wrapped_config_type,
```

Comment:
minor nit: if think we convenient we should avoid reassignment (oh to have `const` in Python) so doing this as one-liner would be nice